mailbox/api/src/main/java/org/apache/james/mailbox/acl/PositiveUserACLDiff.java

```diff
@@ -18,53 +18,36 @@
  ****************************************************************/
 package org.apache.james.mailbox.acl;
 
-import java.util.Map;
 import java.util.stream.Stream;
 
 import org.apache.james.mailbox.model.MailboxACL;
 
 public class PositiveUserACLDiff {
+    private final ACLDiff aclDiff;
 
-    public static PositiveUserACLDiff computeDiff(MailboxACL oldACL, MailboxACL newACL) {
-        return new PositiveUserACLDiff(oldACL, newACL);
+    private PositiveUserACLDiff(ACLDiff aclDiff) {
+        this.aclDiff = aclDiff;
     }
 
-    private final MailboxACL oldACL;
-    private final MailboxACL newACL;
-
-    private PositiveUserACLDiff(MailboxACL oldACL, MailboxACL newACL) {
-        this.oldACL = oldACL;
-        this.newACL = newACL;
+    public static PositiveUserACLDiff computeDiff(ACLDiff aclDiff) {
+        return new PositiveUserACLDiff(aclDiff);
     }
 
     public Stream<MailboxACL.Entry> addedEntries() {
-        Map<MailboxACL.EntryKey, MailboxACL.Rfc4314Rights> oldEntries = oldACL.ofPositiveNameType(MailboxACL.NameType.user);
-
-        return newACL.ofPositiveNameType(MailboxACL.NameType.user)
-            .entrySet()
-            .stream()
-            .filter(entry -> !oldEntries.containsKey(entry.getKey()))
-            .map(entry -> new MailboxACL.Entry(entry.getKey(), entry.getValue()));
+        return aclDiff.addedEntries()
```

Comment:
We can introduce a

```
boolean hasPositiveUserKey(Entry<Key, Rights> ...);
```

 - It would give it a name
 - It would factorize that knowledge.